python/lsst/faro/measurement/TractTableMeasurementTasks.py

```diff
@@ -144,3 +150,123 @@ def run(
+
+    Optional Input (yaml file)
+    ----------
+    Column names specified as str in yaml configuration for TeX task. These are
+    the desired column names to be passed to the calcuation. If you wish to use
+    values other than the default values specified below, add the following e.g.
+    line to the yaml file:
+
+    config.measure.raColumn = "coord_ra_new"
+    """
+
+    columns = DictField(
+        doc="""Columns required for metric calculation. Should be all columns in SourceTable contexts,
+        and columns that do not change name with band in ObjectTable contexts""",
+        keytype=str,
+        itemtype=str,
+        default={"ra": "coord_ra",
+                 "dec": "coord_dec",
+                 }
+    )
+    columnsBand = DictField(
+        doc="""Columns required for metric calculation that change with band in ObjectTable contexts""",
+        keytype=str,
+        itemtype=str,
+        default={"psfFlux": "psfFlux"
+                 }
+    )
+    stellarLocusFitDict = DictField(
+        doc="The parameters to use for the stellar locus fit. The default parameters are examples and are "
+            "not useful for any of the fits. The dict needs to contain xMin/xMax/yMin/yMax which are the "
+            "limits of the initial box for fitting the stellar locus, mHW and bHW are the initial "
+            "intercept and gradient for the fitting.",
+        keytype=str,
+        itemtype=float,
+        default={"xMin": 0.28, "xMax": 1.0, "yMin": 0.02, "yMax": 0.48, "mHW": 0.52, "bHW": -0.08}
+    )
+
+
+class WPerpTableTask(Task):
+    """Class to perform the wPerp calculation on a parquet table data
+    object.
+
+    Parameters
+    ----------
+    None
+
+    Output:
+    ----------
+    wPerp stellar locus metric with defined configuration.
+    """
+
+    ConfigClass = WPerpTableConfig
+    _DefaultName = "wPerpTableTask"
+
+    def run(
+        self, metricName, catalog, currentBands, **kwargs
+    ):
+
+        self.log.info("Measuring %s", metricName)
+
+        # Validation to require that gri bands are in the input set
+        if set(currentBands).issuperset(set(['g', 'r', 'i'])) is False:
+            self.log.warn("Data in gri bands required for wPerp calculation.")
+            return Struct(measurement=Measurement(metricName, np.nan * u.mmag))
+
+        # filter catalog
+        catalog = selectors.applySelectors(catalog,
+                                           self.config.selectorActions,
+                                           currentBands=currentBands)
+
+        # Filter based on configured r-mag limits:
+        # rmag_tmp = (catalog.r_psfFlux.values*u.nJy).to(u.ABmag).value
+        # magfilter = (rmag_tmp < self.config.faint_rmag_cut) &\
+        #             (rmag_tmp > self.config.bright_rmag_cut)
+        # catalog = catalog[magfilter]
+
+        # Create a SkyCoord object to get the extinction:
+        sc = SkyCoord(catalog['coord_ra']*u.deg, catalog['coord_dec']*u.deg)
```

Comment:
Did you run PEP8? There are a few errors.

Reply:
All's well when I ran it. I presume what you're referring to is on an early commit, and was fixed before the PR.